<h3 style="margin:0px">Class: org.jclouds.dynect.v3.DynectApiMockTest (3 methods) </h3><br><pre style="padding:0"><code class="html"><div id="tags"><button id="10"style="border-width:0;padding:0;background:#282B2E"  onclick="filter('10')" data-toggle="tooltip" title="Executes methods or other tests from the same test unit"><kbd id="tag-10"class="label-info"style="display: inline-block;font-size:7pt" >ExecutionTester&nbsp;(3)</kbd></button>&nbsp;<button id="3"style="border-width:0;padding:0;background:#282B2E"  onclick="filter('3')" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution"><kbd id="tag-3"class="label-info"style="display: inline-block;font-size:7pt" >ExceptionVerifier&nbsp;(3)</kbd></button>&nbsp;</div></code></pre><center><button onclick="showBlocks()" style="margin:6px">Update filter <span class="glyphicon glyphicon-refresh" aria-hidden="true"></span> </button></center><br>
<pre class="type-3 type-10 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(expectedExceptions=JobStillRunningException.class,expectedExceptionsMessageRegExp="This session already has a job running") public void test200OnFailureThrowsExceptionWithoutRetryWhenJobRunning() throws IOException, InterruptedException {
  MockWebServer server=new MockWebServer();
  server.enqueue(new MockResponse().setResponseCode(OK.getStatusCode()).setBody(session));
  server.enqueue(new MockResponse().setResponseCode(OK.getStatusCode()).setBody(running));
  server.play();
  DynECTApi api=mockDynectApi(server.getUrl("/").toString());
  try {
    api.getZoneApi().list();
  }
  finally {
    server.shutdown();
  }
}

</code></pre>

<pre class="type-3 type-10 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(expectedExceptions=TargetExistsException.class,expectedExceptionsMessageRegExp="Name already exists") public void test200OnFailureThrowsExceptionWithoutRetryOnNameExists() throws IOException, InterruptedException {
  MockWebServer server=new MockWebServer();
  server.enqueue(new MockResponse().setResponseCode(OK.getStatusCode()).setBody(session));
  server.enqueue(new MockResponse().setResponseCode(OK.getStatusCode()).setBody(targetExists));
  server.play();
  DynECTApi api=mockDynectApi(server.getUrl("/").toString());
  try {
    api.getZoneApi().list();
  }
  finally {
    server.shutdown();
  }
}

</code></pre>

<pre class="type-3 type-10 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(expectedExceptions=JobStillRunningException.class,expectedExceptionsMessageRegExp="Operation blocked by current task") public void test200OnFailureThrowsExceptionWithoutRetryWhenOperationBlocked() throws IOException, InterruptedException {
  MockWebServer server=new MockWebServer();
  server.enqueue(new MockResponse().setResponseCode(OK.getStatusCode()).setBody(session));
  server.enqueue(new MockResponse().setResponseCode(OK.getStatusCode()).setBody(taskBlocking));
  server.play();
  DynECTApi api=mockDynectApi(server.getUrl("/").toString());
  try {
    api.getZoneApi().list();
  }
  finally {
    server.shutdown();
  }
}

</code></pre>

<script>$(document).ready(function() {
  $('pre code').each(function(i, block) {
    hljs.highlightBlock(block);
  });
});</script>
